testutils/src/main/java/com/amplifyframework/testutils/Await.java

```diff
@@ -146,6 +146,9 @@ private Await() {}
                         resultContainer.set(result);
                         latch.countDown();
                     }, error -> {
+                        if (errorContainer.get() != null) {
+                            throw new RuntimeException("Error callback called more than once with error: " + error);
+                        }
```

Comment:
Great addition for catching this type of issues that were previously undetected!

Reply:
Credit to @richardmcclellan for sharing this with me 😄